Mathlib/Topology/Sober.lean

```diff
@@ -112,6 +113,11 @@ theorem IsIrreducible.genericPoint_spec [QuasiSober α] {S : Set α} (hS : IsIrr
     IsGenericPoint hS.genericPoint (closure S) :=
   (QuasiSober.sober hS.closure isClosed_closure).choose_spec
 
+theorem IsIrreducible.genericPoint_spec' [QuasiSober α] {S : Set α}
+    (hS : IsIrreducible S) (hS' : IsClosed S) :
+    IsGenericPoint hS.genericPoint S := by
+  convert hS.genericPoint_spec; exact hS'.closure_eq.symm
+
 @[simp]
 theorem IsIrreducible.genericPoint_closure_eq [QuasiSober α] {S : Set α} (hS : IsIrreducible S) :
```

Comment:
Could you add a prime version of this too, so the fact that when `S` is closed the closure of its generic point equals S.